### ISSUE TITLE
Flyout placement after filepicker

### ISSUE THREAD

**Issue by balikian** [timheuer/callisto#196]
I'm trying to open the flyout to use as a confirmation window after someone clicks a button in the bottom app bar which opens the file picker.  The problem I'm having is after the filepicker is done doing it's thing, I can't seem to position the flyout over the original button.  I've even given than button a name and referenced it directly that way.  It always seems to pop it up in the top left of the window.  I know I should paste some code in here, and I can if there's not an easy answer or if I didn't explain this correctly, but just sort of hoping someone has a quick direction they could point me to.  thanks!

scott

**Comment by timheuer** [comment]
Can you provide a repro of the issue we can try

**Comment by balikian** [comment]
Well, here's some code.  I've tried to make it legible, but who knows!  Thanks for the quick response.  I'm using the flyout in other places and it's working great.  Just in this one instance where we open the filepicker, it pops up in the upper left of the screen.  I've tried to copy and paste code from different files and functions all into one.

private async void SaveToFavoritesClickHDD(object sender, RoutedEventArgs e)
        {

```
        Flyout f = new Flyout();

        Border b = new Border();
        b.Width = 209;
        b.Height = 104;
        b.Name = "addFavoriteContainer";


        Grid grid = new Grid();

        TextBlock tb = new TextBlock();
        tb.Style = Application.Current.Resources["ConfirmationText"] as Style;
        tb.Text = "some text here";

        grid.Children.Add(tb);
        b.Child = grid;
        f.Content = b;
        f.Name = "FlyoutContainer";

        f.Placement = PlacementMode.Top;
        f.PlacementTarget = sender as UIElement; // this is an UI element (usually the sender)

        FileOpenPicker openPicker = new FileOpenPicker();
        openPicker.ViewMode = PickerViewMode.List;
        openPicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
        openPicker.FileTypeFilter.Add(".jpg");
        openPicker.FileTypeFilter.Add(".jpeg");
        openPicker.FileTypeFilter.Add(".gif");
        openPicker.FileTypeFilter.Add(".png");
        IReadOnlyList<StorageFile> files = await openPicker.PickMultipleFilesAsync();
        if (files.Count > 0)
        {
            await ImageFeed.SaveImages(files);
        }

        Storyboard story = new Storyboard();
        story.BeginTime = TimeSpan.FromMilliseconds(2000);
        story.Duration = new Duration(TimeSpan.FromMilliseconds(750));

        FadeOutThemeAnimation fadeOut = new FadeOutThemeAnimation();
        story.Children.Add(fadeOut);
        Storyboard.SetTarget(fadeOut, f);
        if (this.Resources.ContainsKey("Story_ID"))
        {
            this.Resources["Story_ID"] = story;
        }
        else
        {
            this.Resources.Add("Story_ID", story);
        }

        f.IsOpen = true;
        story.Begin();

        story.Completed += storyCompleted;
       }
```

**Comment by timheuer** [comment]
@balikian when I use your code above things are placed fine.  I had to comment out your storyboad because I don't know what XAML that corresponds to.  It is possible that may be conflicting, but using the literal code above minus the storyboard code and commenting out the SaveImages() function, it works as I would expect and the flyout is over the button that invoked it.

**Comment by balikian** [comment]
Thanks Tim.  I tried commenting out the storyboard stuff and it's still misplacing.  I've found if this line is commented out (IReadOnlyList<StorageFile> files = await openPicker.PickMultipleFilesAsync();) that it seems to place it just fine which is why I'd thought it had something to do with the filepicker.  Perhaps when the filepicker gets opened, we lose the window bounds or something.  I don't know when Callisto determines placement which is why I tried setting the PlacementTarget before we get to the filepicker bit, but maybe it doesn't really determine placement until isOpen is set to true?

thanks for your help on this.
